webapp/status_server.py

```diff
@@ -96,23 +96,32 @@ def complete_task():
     'completed': True
   })
 
+
+  water = db.child('groups').child(group).child('resources').child('water').get().val()
+  food = db.child('groups').child(group).child('resources').child('food').get().val()
+  metal = db.child('groups').child(group).child('resources').child('metal').get().val()
   resource = random.randint(0, 2)
   if resource == 0:
```

Comment:
This long if-else can also be reduced.
Use a map structure to determine the resource and a method to update the value (this will make things scalable).
eg UpdateResource(resourceName, valueToAdd)

Reply:
Thanks for your advice! I've just refactored this file accordingly.